SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ExpectedViewDelta.cpp

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Algo/Compare.h"
+#include "ComponentTestUtils.h"
+#include "SpatialView/ViewDelta.h"
+#include "Tests/SpatialView/ExpectedEntityDelta.h"
+#include "Tests/SpatialView/ExptectedViewDelta.h"
+
+using namespace SpatialGDK;
+
+ExpectedViewDelta& ExpectedViewDelta::AddEntityDelta(const Worker_EntityId EntityId, const EntityChangeType ChangeType)
+{
+	EntityDeltas.Add(EntityId, { EntityId, ChangeType == ADD, ChangeType == REMOVE });
+	return *this;
+}
+
+ExpectedViewDelta& ExpectedViewDelta::AddComponentAdded(const Worker_EntityId EntityId, ComponentData Data)
+{
+	EntityDeltas[EntityId].ComponentsAdded.Push(ComponentChange(Data.GetComponentId(), Data.GetUnderlying()));
+	EntityDeltas[EntityId].DataStorage.Push(MoveTemp(Data));
+	return *this;
+}
+
+ExpectedViewDelta& ExpectedViewDelta::AddComponentRemoved(const Worker_EntityId EntityId, const Worker_ComponentId ComponentId)
+{
+	EntityDeltas[EntityId].ComponentsRemoved.Push(ComponentChange(ComponentId));
+	return *this;
+}
+
+ExpectedViewDelta& ExpectedViewDelta::AddComponentUpdate(const Worker_EntityId EntityId, ComponentUpdate Update)
+{
+	EntityDeltas[EntityId].ComponentUpdates.Push(ComponentChange(Update.GetComponentId(), Update.GetUnderlying()));
+	EntityDeltas[EntityId].UpdateStorage.Push(MoveTemp(Update));
+	return *this;
+}
+
+ExpectedViewDelta& ExpectedViewDelta::AddComponentRefreshed(const Worker_EntityId EntityId, ComponentUpdate Update, ComponentData Data)
+{
+	EntityDeltas[EntityId].ComponentsRefreshed.Push(ComponentChange(Update.GetComponentId(), Data.GetUnderlying(), Update.GetEvents()));
+	EntityDeltas[EntityId].DataStorage.Push(MoveTemp(Data));
+	EntityDeltas[EntityId].UpdateStorage.Push(MoveTemp(Update));
+	return *this;
+}
+
+ExpectedViewDelta& ExpectedViewDelta::AddAuthorityGained(const Worker_EntityId EntityId, const Worker_ComponentId ComponentId)
+{
+	EntityDeltas[EntityId].AuthorityGained.Add(AuthorityChange(ComponentId, AuthorityChange::AUTHORITY_GAINED));
+	return *this;
+}
+
+ExpectedViewDelta& ExpectedViewDelta::AddAuthorityLost(const Worker_EntityId EntityId, const Worker_ComponentId ComponentId)
+{
+	EntityDeltas[EntityId].AuthorityLost.Add(AuthorityChange(ComponentId, AuthorityChange::AUTHORITY_LOST));
+	return *this;
+}
+
+ExpectedViewDelta& ExpectedViewDelta::AddAuthorityLostTemporarily(const Worker_EntityId EntityId, const Worker_ComponentId ComponentId)
+{
+	EntityDeltas[EntityId].AuthorityLostTemporarily.Add(AuthorityChange(ComponentId, AuthorityChange::AUTHORITY_LOST_TEMPORARILY));
+	return *this;
+}
+
+ExpectedViewDelta& ExpectedViewDelta::AddDisconnect(const uint8_t StatusCode, FString StatusMessage)
```

Comment:
Super nit - use uint8 if we are using FString. Basically try to use all unreal types or all stl types so we convert in as few places as possible